CondCore/CSCPlugins/src/plugin.cc

```diff
@@ -62,17 +62,17 @@
 
 
 namespace cond {
-  template <> std::shared_ptr<CSCReadoutMapping> deserialize<CSCReadoutMapping>( const std::string& payloadType,
-										   const Binary& payloadData,
-										   const Binary& streamerInfoData ){
+  template <> std::unique_ptr<CSCReadoutMapping> deserialize<CSCReadoutMapping>( const std::string& payloadType,
+                                                                                 const Binary& payloadData,
+                                                                                 const Binary& streamerInfoData ){
     // DESERIALIZE_BASE_CASE( CSCReadoutMapping ); abstract
     DESERIALIZE_POLIMORPHIC_CASE( CSCReadoutMapping, CSCReadoutMappingFromFile );
```

Comment:
the macro DESERIALIZE_POLIMORPHIC_CASE returns a std::shared_ptr<T> .  Are you here converting it in std::unique_ptr<T>?  Is this thing compiling at all?  Or am I missing something?

Reply:
Yes, the return types are changed from shared_ptr to unique_ptr for the deserialize function and everything related to that which is either internal to deserialize or in the call stack up to ESProducer produce functions. When the DESERIALIZE_POLIMORPHIC_CASE is used there is an implicit conversion of the pointer type (and a move to another unique_ptr). Everything in the CMSSW repository compiles (see PR test build and the PR tests all passed). The only reason to have a design where deserialize returns a shared_ptr would be if it or something internal to it actually shares ownership and management of the memory of the allocated object. It currently does not and I cannot imagine ever modifying deserialize to do that in the future. As far as code that uses the deserialize function, it is easy to convert a unique_ptr to a shared_pt if you ever want to do that at any higher level. But it is problematic to ever change from a shared_ptr to a unique_ptr at a higher level. These pointers get passed up all the way to the output of ESProducer produce functions. If the produce functions in ESProducers return shared_ptr's, then there will be problems when we start processing multiple IOVs concurrently. That is the motivation for this PR. I hope that explains it. Let me know if there are more questions or discussion. (I did highlight this in the comments at the top of the PR as I understand this is an interface change in something important and low level).

Reply:
I confirm that these changes would not compile without the corresponding change in CondCore/CondDB/Serialization.h, or a re-impelmentation of the macro.  I can be wrong, but I don't find the file in your list.  
Anyhow, discussing in the context of the validation of a single PR with the title "Prepare for concurrent IOVs" a drastic change in the CondDB interface, like the move from shared_ptr to unique_ptr does not sounds to me appropriate. 
I'm afraid you will have to revert this change.